workflows/detaxizer.nf

```diff
@@ -267,134 +276,222 @@ workflow DETAXIZER {
-            if (blastn[0] == null){
-                blastn[0] = "${projectDir}/assets/NO_FILE1"
+        ch_reference_fasta_with_meta = ch_reference_fasta.map {
+            item -> [['id': "id-fasta-for-makeblastdb"], item]
             }
-            if (blastn[1] == null){
-                blastn[1] = "${projectDir}/assets/NO_FILE2"
+
+        BLAST_MAKEBLASTDB (
+                ch_reference_fasta_with_meta
+        )
+        ch_versions = ch_versions.mix(BLAST_MAKEBLASTDB.out.versions)
+
+        ch_fasta4blastn = PREPARE_FASTA4BLASTN.out.fasta
+            .flatMap { meta, fastaList ->
+                if (fastaList.size() == 2) {
+            // Handle paired-end FASTA files
+                return [
+                    [['id': "${meta.id}_R1", 'single_end': false], fastaList[0]],
+                    [['id': "${meta.id}_R2", 'single_end': false], fastaList[1]]
+                ]
+
+            } else {
+            // Handle single-end FASTA files
+            return [ [ [ 'id': "${meta.id}", 'single_end': true], fastaList ] ]
             }
-            if (blastn[2] == null){
-                blastn[2] = "${projectDir}/assets/NO_FILE3"
+
             }
-            if (filteredblastn[0] == null){
-                filteredblastn[0] = "${projectDir}/assets/NO_FILE4"
+        ch_blastn_db = ch_fasta4blastn.combine(BLAST_MAKEBLASTDB.out.db).map{ it -> [it[2], it[3]]}
+        ch_blastn_db.dump()
+        BLAST_BLASTN (
+            ch_fasta4blastn,
+            ch_blastn_db
+        )
+
+        ch_versions = ch_versions.mix(BLAST_BLASTN.out.versions.first())
+
+        ch_combined_blast = BLAST_BLASTN.out.txt
+                                    .map { meta, path ->
+            [ ['id': meta.id.replaceAll("(_R1|_R2|_R3)", "")], path ]
+        }
+
+
+        ch_combined_blast = ch_combined_blast
+                                .groupTuple(by: [0])
+                                .map { meta, paths -> [ meta, paths.flatten() ] }
+
+        FILTER_BLASTN_IDENTCOV (
+            BLAST_BLASTN.out.txt
+        )
+        ch_versions = ch_versions.mix(FILTER_BLASTN_IDENTCOV.out.versions.first())
+
+        ch_filtered_combined = FILTER_BLASTN_IDENTCOV.out.classified.map {
+            meta, path ->
+            [ ['id': meta.id.replaceAll("(_R1|_R2|_R3)", "")], path ]
+        }
+        ch_filtered_combined = ch_filtered_combined
+            .groupTuple ()
+            .map {
+                meta, paths ->
+                paths = paths.flatten()
+                return [meta, paths]
             }
-            if (filteredblastn[1] == null){
-                filteredblastn[1] = "${projectDir}/assets/NO_FILE5"
+
+        ch_blastn_combined = ch_combined_blast.join(ch_filtered_combined, remainder: true)
+
+        ch_blastn_combined = ch_blastn_combined.map {
+            meta, blastn, filteredblastn ->
+                if (blastn[0] == null){
+                    blastn[0] = "${projectDir}/assets/NO_FILE1"
```

Comment:
I've seen these files and wondered what they are for. Is there a less magic file way? Create one on the fly or use an empty channel?

Reply:
Sadly this is the only solution I found ... I have to handle Input that can be null SOMETIMES (if only one file is present, no need for 3 NO_FILEs anymore as I split up the long and short read input) and this breaks the process. Otherwise I would have to touch the BLAST nf-core module, what I do not want to do at this point. Empty channel is also not an option

Reply:
What about an empty list, i.e. `[]`? This will be accepted as `path(xyz)` when `xyz=[]` afaik. Then you can do in the `script` block `code = xyz ? <code when file exists> : <code when file doesnt exist>`, alternatively for code, return a boolean and evaluate that in your script later.